Functions/Misc/colors

```diff
@@ -82,10 +82,13 @@ for k in ${(k)color}; do color[${color[$k]}]=$k; done
 for k in ${color[(I)3?]}; do color[fg-${color[$k]}]=$k; done
 
 # This is inaccurate, but the prompt theme system needs it.
+# Note: "gray" is US spelling.
 
-color[grey]=${color[black]}
-color[fg-grey]=${color[grey]}
-color[bg-grey]=${color[bg-black]}
+for k in grey gray; do
+  color[$k]=${color[black]}
+  color[fg-$k]=${color[grey]}
```

Comment:
```suggestion
  color[fg-$k]=${color[$k]}
```

Otherwise there'll be a hard-to-catch bug if somebody sorts the values that are iterated over.